lib/spack/spack/test/config.py

```diff
@@ -1861,3 +1861,113 @@ def __call__(self, *args, **kwargs) -> str:  # type: ignore
     include.branch = ""  # type: ignore[union-attr]
     with pytest.raises(spack.error.ConfigError, match="Missing or unsupported options"):
         include.scopes(parent_scope)
+
+
+def test_missing_include_scope_list(mock_missing_include_scopes):
+    """Tests that an included scope with a non existent file/directory
+    is still listed as a scope under spack.config.CONFIG.scopes"""
+    assert "sub_base" in list(
+        spack.config.CONFIG.scopes
+    ), "Missing Optional Scope Missing from Config Scopes"
+
+
+def test_missing_include_scope_writable_list(mock_missing_include_scopes):
+    """Tests that missing include scopes are included in writeable config lists"""
+    assert [x for x in spack.config.CONFIG.writable_scopes if x.name == "sub_base"]
+
+
+def test_missing_include_scope_not_readable_list(mock_missing_include_scopes):
+    """Tests that missing include scopes are not included in readable config lists"""
+    readable_scopes = [x for x in spack.config.CONFIG.readable_scopes if x.name != "sub_base"]
+    assert len(readable_scopes) == 1
+    assert readable_scopes[0].name != "sub_base"
+
+
+def test_missing_include_scope_default_created_as_dir_scope(mock_missing_include_scopes):
+    """Tests that an optional include with no existing file/directory and no yaml extention
+    is created as a directoryscope object"""
+    missing_inc_scope = spack.config.CONFIG.scopes["sub_base"]
+    assert isinstance(missing_inc_scope, spack.config.DirectoryConfigScope)
+
+
+def test_missing_include_scope_yaml_ext_is_file_scope(tmp_path):
+    """Tests that an optional include scope with no existing file/directory and a
+    yaml extension is created as a file scope"""
+    base_scope_dir = tmp_path / "base"
+    scope = spack.config.DirectoryConfigScope("base", str(base_scope_dir))
+    scope.sections["include"] = syaml.syaml_dict(
+        {
+            "include": [
+                {
+                    "name": "sub_base",
+                    "path": str(tmp_path / "sub.yaml"),
+                    "optional": True,
+                    "prefer_modify": True,
+                }
+            ]
+        }
+    )
```

Comment:
(minor) might be worth consolidating this w/missing_include_scope_read_directory into one fixture (which would also help w/ documenting the singlefilescope behavior)